toolchain/check/cpp/import.cpp

```diff
@@ -2329,10 +2314,21 @@ static auto LookupMacro(Context& context, SemIR::NameScopeId scope_id,
   if (!name_str_opt || !IsTopCppScope(context, scope_id)) {
     return nullptr;
   }
-  auto macro_info = context.sem_ir().cpp_macros().find(*name_str_opt);
-  if (macro_info != context.sem_ir().cpp_macros().end()) {
-    return macro_info->second;
+
+  clang::Preprocessor& preprocessor = context.clang_sema().getPreprocessor();
+  clang::IdentifierInfo* identifier_info =
```

Comment:
It'd be nice to do this identifier lookup only once, rather than once here and once in `ClangLookupName`. I'd be happy with a TODO if you'd like to get this landed to unblock further macro work.